src/jaxsim/rbda/contacts/rigid.py

```diff
@@ -194,7 +195,7 @@ def impact_velocity(
                 return nu_post
 
         with data.switch_velocity_representation(VelRepr.Mixed):
-            BW_ν_pre_impact = data.generalized_velocity()
+            BW_ν_pre_impact = BW_nu
```

Comment:
Extracting $\boldsymbol{\nu}$ from `data` has almost zero overhead, I'm not sure that passing it from the outside is worth. This change introduces the risk that the caller (maybe in user code) passes a velocity unaligned with `data`. What do you think?

Reply:
Ok, I agree that this can introduce more risks than performance advantages. I will remove this change in the next commit.

Reply:
Done in 2af4b8b8c07f45e3e64689d7a2799a097d13a609